Add more changes to upgrade guides: masks, panel configuration, action namespaces, action function calls (#7416)
laravel-filament/filament: packages/forms/docs/10-upgrade-guide.md

rm config/forms.php
```

> #### Panel configuration is now in each AppPanelProvider

Reviewer: Belongs in the panels upgrade guide, high impact section. "AppPanelProvider" can just be "panel provider"

Author: Ah I see my confusion here. I will edit this.

IMO would useful to have one upgrade page instead of having to chase down multiple, or at least link them to each other. While Filament supports modular usage, structuring the docs around that instead of an "ordinary" use case involving most of the packages makes it confusing to navigate.

Reviewer: Yeah, the problem is that people who are not using the panel builder at all will then have to skip past all the stuff that isn't relevant to them

- `Closure $get` to `\Filament\Forms\Get $get`
- `Closure $set` to `\Filament\Forms\Set $set`

> #### `TextInput` masks are replaced with Alpine.js masks

Reviewer: ```suggestion
#### `TextInput` masks now use Alpine.js' masking package
```

- `Closure $get` to `\Filament\Forms\Get $get`
- `Closure $set` to `\Filament\Forms\Set $set`

#### `TextInput` masks are replaced with Alpine.js masks

> Filament v2 had a fluent closure mask syntax for managing more complex input masks. In v3 you can use Alpine.js masking instead. So instead of 

Reviewer: ```suggestion
Filament v2 had a fluent mask object syntax for managing input masks. In v3 you can use Alpine.js' masking syntax instead. Please see the [input masking documentation](fields/text-input#input-masking) for more information.
```

> ```php
> ->mask(fn (TextInput\Mask $mask) => $mask
>     ->numeric() // And any other transformations needed
> )
> ```
> You can do:
> ```php
> ->mask(RawJs::make(<<<'JS'
>     $input.startsWith('34') || $input.startsWith('37') ? '9999 999999 99999' : '9999 9999 9999 9999'
> JS))
> ```
> If you were using input masks for money related fields, Alpine already has [builtin helpers](https://alpinejs.dev/plugins/mask#money-inputs) for it:
> ```php
> ->mask(RawJs::make(<<<'JS'
>     $money($input, '.', ',', 2)
> JS))
> ```

Reviewer: ```suggestion
```

- `Closure $get` to `\Filament\Forms\Get $get`
- `Closure $set` to `\Filament\Forms\Set $set`

> #### TextInput numeric masks are replaced with Alpine.js masks

Reviewer: Not just numeric masks, the entire mask object has been replaced. Please use `` for code snippets.

Author: >Please use `` for code snippets.

Three backticks is the proper markdown method for code snippets, and is used throughout the existing docs, so I'm not sure if you meant something else here.

Reviewer: I mean `TextInput` here instead of TextInput, for example

Filament v2 had a fluent closure mask syntax for managing more complex input masks. In v3 you can use Alpine.js masking instead. So instead of 
```
->mask(fn (TextInput\Mask $mask) => $mask
    ->numeric() // And any other transformations needed
)
```
You can do:
```
->mask(RawJs::make(<<<'JS'
    $input.startsWith('34') || $input.startsWith('37') ? '9999 999999 99999' : '9999 9999 9999 9999'
JS))
```
If you were using input masks for money related fields, Alpine already has [builtin helpers](https://alpinejs.dev/plugins/mask#money-inputs) for it:
```
->mask(RawJs::make(<<<'JS'
    $money($input, '.', ',', 2)
JS))
```

> #### Action execution with forms

Reviewer: Doesn't really belong in the forms upgrade guide, should probably be in a new actions upgrade guide file that you'll need to create.
